pyomo/common/dependencies.py

```diff
@@ -563,10 +567,17 @@ def _perform_import(
     import_error = None
     version_error = None
     try:
-        if importer is None:
-            module = importlib.import_module(name)
-        else:
-            module = importer()
+        with warnings.catch_warnings():
```

Comment:
Is this what we want? Hiding warnings seems like it could be a bad idea.

Reply:
It is not actually catching warnings... that turns out to be a (poorly named) context manager that will restore the state of the warnings system on exit (needed because we will sometimes explicitly disable the mapping of warnings to exceptions below).

Reply:
Ahh. Thanks.